compiler/rustc_attr_data_structures/src/version.rs

```diff
@@ -16,8 +17,29 @@ pub struct RustcVersion {
 
 impl RustcVersion {
     pub const CURRENT: Self = current_rustc_version!();
+    pub fn current_configurable() -> Self {
```

Comment:
Suggestion: can you instead name this (and the static below) as `current_overridable()`? `RUSTC_OVERRIDE_VERSION_STRING` is perma-unstable and internal, and we can have a configuration (through `CFG_VERSION` and friends), but `RUSTC_OVERRIDE_VERSION_STRING` will *override* those config options.